app/components/retriable_fast_image/index.tsx

```diff
@@ -33,7 +51,7 @@ export default class RetriableFastImage extends PureComponent<RetriableFastImage
         return (
             <FastImage
                 {...this.props}
-                key={`${this.props.id}-${this.state.retry}`}
+                testID={`${this.props.id}-${this.state.retry}`}
```

Comment:
huh? they key you should keep..
the testID should be part of the prop.

Reply:
The `key`; or `this.props.id` was really only being used for the tests from what I could tell; and I saw a few errors around getting an `undefined` key, and then when I tried to modify it in this `RetriableFastImage` component, it gave the `You attempted to set the key on an object that is meant to be immutable and has been frozen` error.

We can:
- change `key` to `testID`
- update each caller of `<RetriableFastImage ... />` to include a key/id
- Make a new `<ValidatedFastImage ... />` component separate from `<RetriableFastImage />`

Reply:
I've updated to see if it solves this case in the one `<Retriable... />` component, let me know if that works.

Reply:
Ok let me explain.. when a react component updates it's key it
Forces the component to run the full lifecycle, this ensures that the FastImage component gets re-created. That is why we have the key here, every time the retry count changes the component changes its key.

As for the testID if the parent component sets one it should get set in this component and it gives a lot more flexibility.

Reply:
Doh! I just assumed there was a `forceUpdate()` being called somewhere I couldn't see it